teuthology/parallel.py

```diff
@@ -73,20 +78,13 @@ def __enter__(self):
         return self
 
     def __exit__(self, type_, value, traceback):
-        self.group.join()
-
         if value is not None:
             return False
 
-        try:
-            # raises if any greenlets exited with an exception
-            for result in self:
-                log.debug('result is %s', repr(result))
-                pass
-        except Exception:
-            # Emit message here because traceback gets stomped when we re-raise
-            log.exception("Exception in parallel execution")
-            raise
+        # raises if any greenlets exited with an exception
```

Comment:
worth noting that it also blocks until completion, IMO, so readers of __exit__ understand that this is essentially a join().

Also, I think this implies that if there are N tasks, and 1..N-1 are still running, but N has raised, we will not notice the exception until N is visited (last)?  I'm not saying that matters, but it's a thing to be aware of if true.